src/main/scala/apiseed/ConfigurationRepository.scala

```diff
@@ -3,7 +3,16 @@ package apiseed
 import apiseed.model.Configuration
 
 class ConfigurationRepository {
-  val configurations: List[Configuration] = List()
+  var configurations: List[Configuration] = List()
 
   def findById(id: String): Option[Configuration] = configurations.find(_.id == id)
+
+  def delete(id: String): Option[Unit] =
+    findById(id) match {
+      case Some(_) => {
+        configurations = configurations.filterNot(_.id == id)
+        Some(Unit)
+      }
+      case None => None
+    }
```

Comment:
Nella PR precedente ti ho consigliato di usare il pattern matching su `Option` perché in quel caso `Some` e `None` "mappavano" su cose diverse, addirittura appartenenti ad un diverso _type_.
In questo caso, invece, `Some` resta `Some` e `None` resta `None`: quindi stiamo facendo un `map()` (vedi [scaladoc](http://www.scala-lang.org/api/2.12.3/scala/Option.html#map[B](f:A=>B):Option[B])).